jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -598,6 +730,26 @@ def p2sh_p2wpkh_sign(tx, i, priv, amount, hashcode=SIGHASH_ALL, usenonce=None):
     txobj["ins"][i]["txinwitness"] = [sig, pub]
     return serialize(txobj)
 
+def p2wpkh_sign(tx, i, priv, amount, hashcode=SIGHASH_ALL, usenonce=None):
```

Comment:
Can actually merge `p2sh_p2wpkh_sign` and `p2sh_p2wpkh` to avoid code re-use.  The sighash is the same, only difference is whether to place the p2wpkh witness program in scriptsig for p2sh-p2wpkh as redeemscript or leave it blank for bare p2wpkh

Reply:
Excellent point, thanks. Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/268/commits/0b558247dd12971d842136724862672c3c204f24